### PR TITLE
Allow developer to specify deprecation for preference

### PR DESCRIPTION
**Pull request by Vegann**

### CODE REVIEW

File: core/lib/spree/core/engine.rb [hunk 1]
```diff
@@ -129,6 +129,13 @@ class Engine < ::Rails::Engine
         Migrations.new(config, engine_name).check
       end
 
+      initializer 'spree.core.checking_deprecated_preferences' do
+        require 'active_support'
```
Comment (Vegann) [inline core/lib/spree/core/engine.rb:133]: I'll remove this, I was testing ActiveSupport::Deprecation

File: core/app/models/spree/app_configuration.rb [hunk 2]
```diff
@@ -43,7 +43,7 @@ class AppConfiguration < Preferences::Configuration
     preference :checkout_zone, :string, default: nil # replace with the name of a zone if you would like to limit the countries
     preference :company, :boolean, default: false # Request company field for billing and shipping addr
     preference :currency, :string, default: 'USD'
-    preference :default_country_id, :integer
+    preference :default_country_id, :integer, deprecated:  'Please use default_country_iso instead. default_country_id will be removed in Spree 5.0'
```
Comment (damianlegawiec) [inline core/app/models/spree/app_configuration.rb:46]: @Vegann we don't have `default_country_iso` preference yet :)

File: backend/lib/spree/backend/engine.rb [hunk 3]
```diff
@@ -11,6 +11,12 @@ class Engine < ::Rails::Engine
       initializer 'spree.params.filter' do |app|
         app.config.filter_parameters += [:password, :password_confirmation, :number]
       end
+
+      initializer 'spree.core.checking_deprecated_preferences' do
+        Spree::Backend::Config.deprecated_preferences.each do |pref|
+          warn "[DEPRECATION] #{pref[:name]} is deprecated #{pref[:message]}"
```
Comment (damianlegawiec) [inline backend/lib/spree/backend/engine.rb:17]: @Vegann what will be the value of the name? the preference key? if yes we need to point them to which gem preference is that, it should be `Spree::Backend::Config[:some_preference]`

**Review by tomurb — APPROVED** [review]

**Comment by tomurb** [comment]
It got into my feed today, I think it could be used in `#deprecated_preferences` - https://everydayrails.com/2021/07/31/rails-custom-deprecation-warnings.html
```
  ActiveSupport::Deprecation.warn(
    "#process_widget is deprecated. " \
    "Use #send_widget_to_processor instead."
  )
 ```

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**